qt-bugs@ issue : 
bugs.kde.org number :
applied: no
author: Dirk Mueller <mueller@kde.org>

gcc 4.3 produces a "3.*" buildkey. That's oh so wrong.


--- configure
+++ configure
@@ -3140,15 +3140,15 @@ case "$COMPILER" in
 g++*)
     # GNU C++
     QMAKE_CONF_COMPILER=`grep "QMAKE_CXX[^_A-Z0-9a-z]" $QMAKESPEC/qmake.conf | sed "s,.* *= *\(.*\)$,\1,"`
-    COMPILER_VERSION=`${QMAKE_CONF_COMPILER} --version 2>/dev/null`
+    COMPILER_VERSION=`${QMAKE_CONF_COMPILER} --version 2>/dev/null | sed 's,^[^0-9]*,,g'`
     case "$COMPILER_VERSION" in
-    *2.95.*)
+    2.95.*)
 	COMPILER_VERSION="2.95.*"
 	;;
-    *3.*)
+    3.*)
 	COMPILER_VERSION="3.*"
 	;;
-    *4.*)
+    4.*)
 	COMPILER_VERSION="4"
 	;;
     *)
